filament/src/details/Renderer.cpp

```diff
@@ -417,6 +417,9 @@ void FRenderer::render(FView const* view) {
 
     assert_invariant(mSwapChain);
 
+    ASSERT_PRECONDITION(!view->hasInstancedStereo() || mEngine.getDriverApi().isStereoSupported(),
+            "Stereo rendering is not supported.");
+
```

Comment:
this should be checked in `setStereoscopicOptions` instead.

Reply:
I thought about that, but all of our `View.setXYZOptions` methods are `noexcept`. Are we okay breaking that convention for `setStereoscopicOptions`?

Reply:
yes, the noexcept predated the feature levels. I think that's the best place for these checks.

The original idea was that these setters would never fail because they would just "clamp" the parameters to something that works. But for feature levels I think we should assert/throw.